internal/app/tfsec/rules/azure/storage/delete_retention_policy_rule.go

```diff
@@ -0,0 +1,232 @@
+					account_replication_type = "LRS"
+					min_tls_version          = "TLS1_2"
+				  
+					network_rules {
+					  default_action             = "Deny"
+					  ip_rules                   = ["100.0.0.1"]
+					  virtual_network_subnet_ids = [azurerm_subnet.example.id]
+					}
+				  
+					tags = {
+					  environment = "staging"
+					}
+				   blob_properties {
+					   delete_retention_policy = 1
+				   } 
+				  }
+				`, `
+				resource "azurerm_resource_group" "example" {
+					name     = "example-resources"
+					location = "West Europe"
+				  }
+				  
+				  resource "azurerm_virtual_network" "example" {
+					name                = "virtnetname"
+					address_space       = ["10.0.0.0/16"]
+					location            = azurerm_resource_group.example.location
+					resource_group_name = azurerm_resource_group.example.name
+				  }
+				  
+				  resource "azurerm_subnet" "example" {
+					name                 = "subnetname"
+					resource_group_name  = azurerm_resource_group.example.name
+					virtual_network_name = azurerm_virtual_network.example.name
+					address_prefixes     = ["10.0.2.0/24"]
+					service_endpoints    = ["Microsoft.Sql", "Microsoft.Storage"]
+				  }
+				  
+				  resource "azurerm_storage_account" "example" {
+					name                = "storageaccountname"
+					resource_group_name = azurerm_resource_group.example.name
+				  
+					location                 = azurerm_resource_group.example.location
+					account_tier             = "Standard"
+					account_replication_type = "LRS"
+					min_tls_version          = "TLS1_2"
+				  
+					network_rules {
+					  default_action             = "Deny"
+					  ip_rules                   = ["100.0.0.1"]
+					  virtual_network_subnet_ids = [azurerm_subnet.example.id]
+					}
+				  
+					tags = {
+					  environment = "staging"
+					}
+				   blob_properties {
+					   delete_retention_policy = 365
+				   } 
+				  }
+				`}, Links: []string{
+				"https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account#delete_retention_policy",
+				"https://docs.microsoft.com/en-us/azure/storage/blobs/soft-delete-blob-enable?tabs=azure-CLI"},
+		},
+		Provider:        provider.AzureProvider,
+		RequiredTypes:   []string{"resource"},
+		RequiredLabels:  []string{"azurerm_storage_account"},
+		DefaultSeverity: severity.Low,
+		CheckFunc: func(set result.Set, resourceBlock block.Block, module block.Module) {
+			if resourceBlock.IsResourceType("azurerm_storage_account") {
+				if resourceBlock.MissingChild("blob_properties") {
+					return
+				}
+				resourceBlock = resourceBlock.GetBlock("blob_properties")
+
+				if resourceBlock.MissingChild("delete_retention_policy") {
+					return
+				}
+				deleteRetention := resourceBlock.GetAttribute("delete_retention_policy")
+				if deleteRetention.IsNotNil() && deleteRetention.IsNumber() {
+					if deleteRetention.GreaterThan(365) || deleteRetention.LessThanOrEqualTo(0) {
```

Comment:
We don't need to check the value is between 1-365, the azurerm provider will do this for us. We need to check that a retention policy is specified at all.